### ISSUE TITLE
DEV Error does not clear after drive replacement.

### ISSUE THREAD

**Issue by cyberlink1** [rockstor/rockstor-core#2292]
I recently had a drive go bad in one of my pools, the bad block created a CSum error on one of the files.

I replaced the drive using btrfs replace, then did a scrub on the drive. BTRFS replace actually copied the csum error but doing a scrub identified the file that was affected. After which I removed the file and copied a new version over. This eliminated the csum error on the file and cleared out the drive issue.
![Rockstor-Scrubs](https://user-images.githubusercontent.com/12261936/113478743-6f44b380-9450-11eb-9f23-f23778de4628.png)
![rockstor-scrub-report](https://user-images.githubusercontent.com/12261936/113478744-6f44b380-9450-11eb-9b9e-a9ef3d255db6.png)

While the issue is fixed and the drive is replaced, rockstor still shows device errors on the pool. It is like it did not clear it out when I fixed the issue.

![rockstor-errors](https://user-images.githubusercontent.com/12261936/113478745-6fdd4a00-9450-11eb-94b3-3c77688cd3f9.png)

This appears to be a bug and it would be nice to see that my NAS is in good health now that it is.

**Comment by phillxnet** [comment]
@cyberlink1 Thanks for the report.

Not a bug as these stats are extracted directly from the dev stats command in btrfs. The are accumulative, i.e. the show the history of errors for a given device. To double check do a btrfs dev stats on that device to see if the command line agrees. And if it does you can execute the specified command to clear the current tally of past errors.

Do report here if you find the dev stats command on the dev/pool do not agree with the Web-UI. Fixing a prior issue does not clear this 'increasing only' tally. Only an intentional clearing of the tally will return it to zero. This is by design within btrfs and we simply reflect this; hopefully.

Give us the output of the dev stats command from the current state to be sure we are properly reflecting this and if we are then this issue should be closed. A feature request to intergrate this clearing of historic data would the the way forward then rather than our little text to indicate how to clear the historic data.

Cheers.

**Comment by phillxnet** [comment]
The text I'm referencing is the "Cumalative pool errors per device ..." bit and the suggested command to reset it. All the red messages are simply keyed from there having been errors and them not having been reset. Not that their cause has now been resolved. It's the current way btrfs works for these errors and has come up on the btrfs mailing list a few times. It's why I used that wording and I wanted to keep the wording as short as possible to help with translations etc.

We probably need better wording here or a link to an as yet non existent doc entry that explains the cumulative nature of these errors.

**Comment by cyberlink1** [comment]
Ok, did not realize I had to manually reset the device stats after the scrub of the device showed clean.

Guess this can be closed. Ill add it to my notes.

**Comment by phillxnet** [comment]
@cyberlink1 Good point though. We could do with a pull request on the docs to explain this, if you fancy having a go. And we could then link to that section (via another pr on core) within the Web-UI as it's obviously a confusing scenario that we are not properly helping with within the Web-UI. Feel free to create the appropriate issues in the docs and core repo for these if you fancy given you interaction here has highlight the potential miss-understanding.

We are trying, within the Web-UI to present the magic of btrfs and add as little as possible ourselves. There is after all plenty enough magic already for folks to be getting along with. In fact far more there than we currently present: but we are working on that.

Cheers.
